### ISSUE TITLE
dynatrace_document destroy is corrupting statefile

### ISSUE THREAD

**Issue by jassi-singh-wig** [dynatrace-oss/terraform-provider-dynatrace#614]
**Describe the bug**

New Dashboard on deletion is getting removed from dynatrace, but remains in statefile and throws error

.....dynatrace_document.dashboard["Health_Check.json"]: Destroying... [id=99ff7acf-8634-4329-bfc1-15f3fab024b3]
```
Error: API request HTTP DELETE https://xxx00000.apps.dynatrace.com/platform/document/v1/trash/documents/99ff7acf-8634-4329-bfc1-15f3fab024b3 failed with status code 403: {"error":{"code":403,"message":"Forbidden","details":{"errorRef":"ca4f059c-016f-4cc4-a2b5-a7c5b826170f"}}}
```
**To Reproduce**
1. Create a new dashboard via terraform using dynatrace_document
2. Remove the dashboard json file from setup, so that it destroys the dashboards from dynatrace
3. Terrafrom plan is ok  `# ...dynatrace_document.dashboard["Health_Check.json"] will be destroyed`
5. Terraform apply throws error `"../platform/document/v1/trash/documents/99ff7acf-8634-4329-bfc1-15f3fab024b3 failed with status code 403:"`
6. Although on checking Dynatrace UI, we can see dashboard is no longer available, but the same exists in statefile

**Expected behavior**
Dashboard document should be removed from the statefile on removal
No errors should be thrown

**Additional context**
- Installing dynatrace-oss/dynatrace v1.70.0...
- Installed dynatrace-oss/dynatrace v1.70.0 (signed by a HashiCorp partner, key ID C8693F64E6CB1026)
```
resource "dynatrace_document" "dashboard" {
  for_each = local.dashboards
  type = "dashboard"
  name = xyzzy
  content = ....
}

resource "dynatrace_direct_shares" "read_write" {
  for_each = local.dashboards
  document_id = dynatrace_document.dashboard[each.key].id
  access      = "read-write"
  recipients {
    recipient {
      id   = ..
      type = "group"
    }
  }
}

resource "dynatrace_direct_shares" "read" {
  for_each = local.dashboards
  document_id = dynatrace_document.dashboard[each.key].id
  access      = "read"
  recipients {
    recipient {
      id   =  ..
      type = "group"
    }
  }
}

```

**Comment by Reinhard-Pilz-Dynatrace** [comment]
Thanks @jassi-singh-wig for reporting this issue. You've stumbled across a regression bug after we've migrated the client for documents to a common repository. That's why well known HTTP error codes like 404 didn't get recognized correctly anymore.

This bug will be fixed with the next release of the provider.

**Comment by jasswig** [comment]
@Reinhard-Pilz-Dynatrace is the fix released in [1.171.0](https://github.com/dynatrace-oss/terraform-provider-dynatrace/releases/tag/v1.71.0) ?, if not, when is the fixed released expected to be released?

**Comment by Reinhard-Pilz-Dynatrace** [comment]
@jasswig no, v1.71 doesn't contain the fix yet. We will likely push out v1.72 before holiday break on Friday. That version will contain the fix.
